src/analysis_file.py

```diff
@@ -0,0 +1,139 @@
+
+        # Convert codes to their string/matrix values
+        for td in data:
+            analysis_dict = dict()
+            for plan in PipelineConfiguration.RQA_CODING_PLANS + PipelineConfiguration.SURVEY_CODING_PLANS:
+                for cc in plan.coding_configurations:
+                    if cc.analysis_file_key is None:
+                        continue
+
+                    if cc.coding_mode == CodingModes.SINGLE:
+                        analysis_dict[cc.analysis_file_key] = \
+                            cc.code_scheme.get_code_with_code_id(td[cc.coded_field]["CodeID"]).string_value
+                    else:
+                        assert cc.coding_mode == CodingModes.MULTIPLE
+                        show_matrix_keys = []
+                        for code in cc.code_scheme.codes:
+                            show_matrix_keys.append(f"{cc.analysis_file_key}{code.string_value}")
+
+                        for label in td.get(cc.coded_field, []):
+                            code_string_value = cc.code_scheme.get_code_with_code_id(label['CodeID']).string_value
+                            analysis_dict[f"{cc.analysis_file_key}{code_string_value}"] = Codes.MATRIX_1
+
+                        for key in show_matrix_keys:
+                            if key not in analysis_dict:
+                                analysis_dict[key] = Codes.MATRIX_0
+            td.append_data(analysis_dict,
+                           Metadata(user, Metadata.get_call_location(), TimeUtils.utc_now_as_iso_string()))
+
+        # Set consent withdrawn based on presence of data coded as "stop"
+        ConsentUtils.determine_consent_withdrawn(
+            user, data, PipelineConfiguration.RQA_CODING_PLANS + PipelineConfiguration.SURVEY_CODING_PLANS,
+            consent_withdrawn_key
+        )
+
+        # Fold data to have one respondent per row
+        to_be_folded = []
+        for td in data:
+            to_be_folded.append(td.copy())
+
+        folded_data = FoldTracedData.fold_iterable_of_traced_data(
+            user, data, fold_id_fn=lambda td: td["uid"],
+            equal_keys=equal_keys, concat_keys=concat_keys, matrix_keys=matrix_keys, bool_keys=bool_keys,
+            binary_keys=binary_keys
+        )
+
+        # Fix-up _NA and _NC keys, which are currently being set incorrectly by
+        # FoldTracedData.fold_iterable_of_traced_data when there are multiple radio shows
+        # TODO: Update FoldTracedData to handle NA and NC correctly under multiple radio shows
+        for td in folded_data:
+            for plan in PipelineConfiguration.RQA_CODING_PLANS + PipelineConfiguration.SURVEY_CODING_PLANS:
+                for cc in plan.coding_configurations:
+                    if cc.analysis_file_key is None:
+                        continue
+
+                    if cc.coding_mode == CodingModes.MULTIPLE:
+                        if td.get(plan.raw_field, "") != "":
```

Comment:
If using the Text field from Rapid Pro rather than the value field then this needs to be `if plan.raw_field in td` (see https://github.com/AfricasVoices/Project-OCHA/pull/69)

Reply:
updated thanks.